### ISSUE TITLE
dnscrypt keep shutting down

### ISSUE THREAD

**Issue by elmakong** [DNSCrypt/dnscrypt-proxy#447]
when i check the log (windows 10 x64) i keep getting this message

> The DNSCrypt client proxy service terminated unexpectedly.  It has done this 3 time(s).

whats wrong?

**Comment by elmakong** [comment]
is it possible because the load balancing flag `lb_strategy = 'fastest'` was set? because when i try to comment that line, it doesn't seem to be crashing anymore

**Comment by iWARR** [comment]
I have lb_strategy = 'fastest' on Win8.1 and everything is fine.

Also notice this https://github.com/jedisct1/dnscrypt-proxy/issues/454 tip.
Looks like Win7 have less copatibility with the rescent versions. This OS is too obsolete today, as well.
But Win8/8.1/Win10 are more "close" versions, instead of Win7/Win10 comparison.

**Comment by elmakong** [comment]
i dont use win7 anymore..also, the problem is not "service wont start"...it will start, but after some time, it just shut down by itself.

**Comment by stylemessiah** [comment]
Confirming that setting fastest in lb_strategy causes service to die - Windows 10 x64 LTSB

**Comment by DRSDavidSoft** [comment]
I have the same issue - service is dying out on Windows Server 2016

**Comment by jedisct1** [comment]
Unless someone has a solution, or a relevant stack trace, or a reliable way to reproduce this, we can't expect this to be solved.

**Comment by DRSDavidSoft** [comment]
@jedisct1 How debugging mode or logs can be enabled on Windows?

Can it run in console mode instead of a background service?

**Comment by jedisct1** [comment]
Run it from the Powershell console, just by typing:

```
dnscrypt-proxy
```

And don't close the Window.

**Comment by DRSDavidSoft** [comment]
@jedisct1 Quick question though, does it matter if it's ran from Cmd.exe or Powershell.exe?

**Comment by jedisct1** [comment]
I haven't used cmd.exe for a long time, but It shouldn't matter at all.

**Comment by KrasnayaPloshchad** [comment]
You can copy the following codes into Notepad, then save as dnscrypt-proxy.bat, put alongside the main program, then it would reproduce if you double click this batch file.
```
@set @_cmd=1 /*
@echo off

@CD /d %~dp0
dnscrypt-proxy -config dnscrypt-proxy.toml
```

**Comment by DRSDavidSoft** [comment]
@KrasnayaPloshchad Interesting, I always used `cd /d %0\..` to switch to the current directory. Out of curiosity, what does `@set @_cmd=1 /*` do? And why not put `@echo off` in the beginning?

BTW, I currently do it with a shortcut (.lnk file) rather than batch scripts.

@jedisct1 I haven't captured the service dying out yet. I'll post a new comment when it happens.

**Comment by KrasnayaPloshchad** [comment]
OK, I dropped `@set @_cmd=1 /*` and `@echo off` in the beginning, but this problem still affect me sometimes.

**Comment by B00ze64** [comment]
@DRSDavidSoft `@set @_cmd=1 /*` is used in the default proxy scripts to hide the batch commands from the javascript call to cscript, when re-running the script to gain admin privileges. Neat trick.

**Comment by DRSDavidSoft** [comment]
@B00ze64 Wow, that seems interesting. Not to get carried away, but what is `cscript` doing here, and how does this workaround work to elevate the script? I'm eager to know!

**Comment by darkclip** [comment]
I have the same problem. I'm running it on centos(x86) and debian(arm), it just can not keep running for more than half a hour, no error messages. However one time I got the flowing stacktrace, hope that's helpful.


`panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x374854]

goroutine 191 [running]:
main.(*Proxy).exchangeWithTCPServer(0x11e0a014, 0x11d6adc0, 0x11d6ae01, 0x12080800, 0x100, 0x100, 0x1216a630, 0xc, 0xc, 0x100, ...)
        /home/travis/gopath/src/github.com/jedisct1/dnscrypt-proxy/dnscrypt-proxy/proxy.go:225 +0xd4
main.(*Proxy).processIncomingQuery(0x11e0a014, 0x11d6adc0, 0x42128f, 0x3, 0x42128f, 0x3, 0x11e59902, 0x20, 0x1000, 0x120be328, ...)
        /home/travis/gopath/src/github.com/jedisct1/dnscrypt-proxy/dnscrypt-proxy/proxy.go:295 +0x998
main.(*Proxy).tcpListener.func1(0x120ea278, 0x11e0a014)
        /home/travis/gopath/src/github.com/jedisct1/dnscrypt-proxy/dnscrypt-proxy/proxy.go:184 +0x1f0
created by main.(*Proxy).tcpListener
        /home/travis/gopath/src/github.com/jedisct1/dnscrypt-proxy/dnscrypt-proxy/proxy.go:171 +0xa8`

**Comment by jedisct1** [comment]
Thanks for the backtrace!

I think that was fixed by a1c8012fc62c2a0998a96c298addfce896aa2329
